### PR TITLE
[make] separate generated gramlib ml files from mli files (fix #10864)

### LINKED ISSUE #10864: Parallel build with 8.10 release results in error about missing gramlib/.pack/gramlib.cmi
#### Description of the problem

Compiling from the release tarball for 8.10.0 using `./configure` and `make -j8` reliably produces the following error about a missing interface for gramlib:

```
OCAMLC    interp/numTok.mli
OCAMLOPT  gramlib/.pack/gramlib.ml
OCAMLC    gramlib/.pack/gramlib.ml
OCAMLC    tactics/dnet.mli
OCAMLC    tactics/dn.mli
OCAMLC    vernac/attributes.mli
OCAMLC    vernac/locality.mli
OCAMLC    stm/spawned.mli
OCAMLC    stm/dag.mli
File "gramlib/.pack/gramlib.ml", line 1:
Error: Could not find the .cmi file for interface gramlib/.pack/gramlib.cmi.
make[1]: *** [gramlib/.pack/gramlib.cmx] Error 2
make[1]: *** Waiting for unfinished jobs....
OCAMLC    stm/tQueue.mli
make: *** [submake] Error 2
```

The build works fine for me without parallelism, but of course is much slower.

#### Coq Version

8.10 on macOS

### PR DESCRIPTION
**Pull request by gares**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -104,12 +104,15 @@ EXISTINGMLI := $(call find, '*.mli')
 GENMLGFILES:= $(MLGFILES:.mlg=.ml)
 # GRAMFILES must be in linking order
 GRAMFILES=$(addprefix gramlib/.pack/gramlib__,Ploc Plexing Gramext Grammar)
-GRAMMLFILES := $(addsuffix .ml, $(GRAMFILES)) $(addsuffix .mli, $(GRAMFILES))
-GENGRAMFILES := $(GRAMMLFILES) gramlib/.pack/gramlib.ml
-GENMLFILES:=$(LEXFILES:.mll=.ml) $(YACCFILES:.mly=.ml) $(GENMLGFILES)  ide/coqide_os_specific.ml kernel/copcodes.ml kernel/uint63.ml
+GRAMMLFILES := $(addsuffix .ml, $(GRAMFILES))
+GRAMMLIFILES := $(addsuffix .mli, $(GRAMFILES))
+GENGRAMMLFILES := $(GRAMMLFILES) gramlib/.pack/gramlib.ml
```
Comment (SkySkimmer) [inline Makefile]: Anyone remember why `gramlib/.pack/gramlib.ml` isn't in `GRAMMLFILES`?
If someone does we should add a comment.

**Review by SkySkimmer — APPROVED** [review]
Looks OK, did you confirm that it fixes the bug?

**Comment by gares** [comment]
@SkySkimmer I think I did what you suggested but things break badly... investigating

**Comment by SkySkimmer** [comment]
It was about the same thing, but I took the opportunity to clean up unused exported variables so let me repush.

**Comment by gares** [comment]
ok then, I hope this is it.

**Comment by gares** [comment]
@vbgl this is to be backported

**Comment by SkySkimmer** [comment]
Squashed

Should I keep the assignment when some of the diff is from me?

**Comment by gares** [comment]
Maybe the release manager can merge and backport

**Pull request merged**